### PR TITLE
Add Alias Functions

### PR DESCRIPTION
**Pull request by Thomas--S**
* register_alias
* register_alias_force

Allows to alias the stairsplus nodes.

### CODE REVIEW

**Comment by sofar** [comment]
Can you describe *why* this is needed? The patch looks OK, but glancing over it I have no idea why you're making the change.

**Comment by Thomas--S** [comment]
In my worlds, I want to force-alias building_blocks:fakegrass to default:dirt_with_grass.
So, I register the stairsplus nodes for the default grass, but then I need to alias the fakegrass to normal grass.

**Comment by sofar** [comment]
I still don't understand why you want to do those things, is there a simpler example? Why would you specifically need to modify moreblocks to accomplish this? 

It is really weird for a mod to assist in overriding it's own items in this way.

**Comment by Thomas--S** [comment]
Another example:
A mod registers an item called `testmod:itemname`. Also the stairsplus nodes for the item will be registered.
After a few days, the mod author decides to call the item `testmod:item_name`. (These things happen often, e.g. because of a typo).
The mod will then do this: `minetest.register_alias("testmod:itemname", "testmod:item_name")`.
But what happens with the stairsplus nodes?
Option 1 (without this PR): The mod has to register 45 aliases by hand for backwards compatibility.
Option 2 (with this PR): The mod uses the `stairsplus:register_alias_all to achieve the same effect as above, but in one line.

So, basically, this PR prevents all the mods that use stairsplus from reinventing the wheel when they want to register aliases for the stairsplus nodes.

**Comment by sofar** [comment]
Ahh, that makes sense now.

**Comment by pithydon** [comment]
I may need to use this in my morerainbows mod to support 0.4.15.
+1

**Comment by Thomas--S** [comment]
@sofar Could you please merge this maybe? Thanks in advance.

**Comment by Thomas--S** [comment]
Fixed a bug that caused wrong items to be dropped. This was introduced by scope problems. It is now avoided by the use of the `stairsplus.copytable` function.
